python/ray/util/tracing.py

```diff
@@ -0,0 +1,392 @@
+from contextlib import contextmanager
+from functools import wraps
+import inspect
+import os
+
+from opentelemetry import context, trace
+from opentelemetry.context.context import Context
+from opentelemetry.util import types
+from opentelemetry.util.types import Attributes
+from opentelemetry import propagators
+from opentelemetry.trace.propagation.textmap import DictGetter
+
+from typing import (
+    Any,
+    cast,
+    Callable,
+    Dict,
+    Generator,
+    MutableMapping,
+    Optional,
+    Sequence,
+    Union,
+)
+
+from ray.runtime_context import get_runtime_context
+from ray.util.inspect import is_function_or_method
+import ray.worker
+
+_nameable = Union[str, Callable[..., Any]]
+is_tracing_enabled = os.getenv("RAY_TRACING_ENABLED", "False").lower() in ["true", "1"]
```

Comment:
_global_is_tracing_enabled = None
def is_tracing_enabled() -> bool:
    if _global_is_tracing_enabled is None:
        _global_is_tracing_enabled = os.getenv(...)
    return _global_is_tracing_enabled